lib/dart.yaml

```diff
@@ -1,194 +1,209 @@
-    - map-keys-ordering
-    - match-positional-field-names-on-assignment
-    - member-ordering:
-        order:
-          - public-fields
-          - private-fields
-          - public-getters
-          - private-getters
-          - public-setters
-          - private-setters
-          - constructors
-          - close-method
-          - dispose-method
-          - public-methods
-          - private-methods
-        widgets-order:
-          - constructor
-          - named-constructor
-          - const-fields
-          - final-fields
-          - init-state-method
-          - did-change-dependencies-method
-          - did-update-widget-method
-          - dispose-method
-          - private-methods
-          - overriden-public-methods
-          - build-method
-    - move-records-to-typedefs:
-        min-fields: 3
-        min-occurences: 2
-    - move-variable-closer-to-its-usage
-    - move-variable-outside-iteration
-    - newline-before-return
-    - no-boolean-literal-compare
-    - no-empty-block
-    - no-equal-arguments:
-        ignored-parameters:
-          - height
-          - width
-    - no-equal-conditions
-    - no-equal-nested-conditions
-    - no-equal-switch-case
-    - no-equal-switch-expression-cases
-    - no-equal-then-else
-    - no-magic-number:
-        allowed: [ -1, 0, 1 ]
-    - no-object-declaration
-    - prefer-addition-subtraction-assignments
-    - prefer-any-or-every
-    - prefer-correct-callback-field-name:
-        name-pattern: ^on[A-Z]+
-    - prefer-correct-error-name:
-        allowed-names: [ error, exception, failure ]
-    - prefer-correct-identifier-length:
-        max-identifier-length: 30
-        min-identifier-length: 2
-    - prefer-correct-stream-return-type
-    - prefer-declaring-const-constructor:
-        ignore-abstract: true
-    - prefer-early-return:
-        ignore-if-case: true
-    - prefer-explicit-function-type
-    - prefer-explicit-type-arguments
-    - prefer-immediate-return
-    - prefer-match-file-name:
-        exclude:
-          - "test/**"
-          - "lib/**/*wm.dart"
-    - prefer-moving-to-variable:
-        allowed-duplicated-chains: 3
-    - prefer-named-boolean-parameters:
-        ignore-single: true
-        ignore-single-boolean: false
-    - prefer-null-aware-spread
-    - prefer-parentheses-with-if-null
-    - prefer-public-exception-classes
-    - prefer-return-await
-    - prefer-simpler-patterns-null-check
-    - prefer-switch-with-enums
-    - prefer-test-matchers
-    - prefer-typedefs-for-callbacks
-    - prefer-unique-test-names
-    - prefer-wildcard-pattern
-    # Style
-    - unnecessary-trailing-comma: false
+    rules:
+        # https://dcm.dev/docs/rules/common
+        - arguments-ordering:
+              last:
+                  - body
+                  - child
+                  - children
+        - parameters-ordering:
+              required: "first"
+              default: "last"
+              super: "last"
```

Comment:
This order of parameters inside class constructors.

Note: I didn't find:
1. how to disable alphabetical order;
2. how to set the order of anonymous fields that are not referred to class fields
Example: 
```
ElementaryWidget({
    super.key,
    /// For this parameter
    WidgetModelFactory wmFactory = defaultWMFactory,
})

```

Reply:
> 1. how to disable alphabetical order;
> 2. how to set the order of anonymous fields that are not referred to class fields

1) I think alphabetical sorting is OK, it's easier to navigate the code that way. 
2) Isn't "default" option responsible for this? https://dcm.dev/docs/rules/common/parameters-ordering/

Reply:
2. Yes, but in our projects such parameters often come after `super`, and parameters with default values, which are specified through `this`, come before.

Reply:
> I think alphabetical sorting is OK, it's easier to navigate the code that way.

To be honest, I used to order arguments by semantic. E.g.:
```dart
  const SomeWidget({
    required this.id,
    required this.title,
    required this.description,
    required this.closeIcon,
    super.key,
  });
 ```

With alphabetic order it'll be look like this:
```dart
const SomeWidget({
      required this.closeIcon,
      required this.description,
      required this.id,
      required this.title,
      super.key,
    });
```

This is not a big problem, but it can be inconvient in some cases

Reply:
> To be honest, I used to order arguments by semantic. E.g.:

I agree, that's why I wanted to turn off the alphabetical order